p2p/service/sim.go

```diff
@@ -33,11 +35,15 @@ type dht interface {
 type Node struct {
 	sim *Simulator
 	node.Node
-	dht dht
+	dht      dht
+	sndDelay uint32
+	rcvDelay uint32
 }
 
 // New Creates a p2p simulation by providing nodes as p2p services and bridge them.
 func NewSimulator() *Simulator {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
you won't need this here if you set the randomness outside. also this changes the global binary random state (its kind of a singleton) and it might be changed by others.

Reply:
Create a new `rand` (using `rand.New()`). instead of using global system.
Also print the seed so tests could be possibly recreated.